tests/components/azure_event_hub/test_init.py

```diff
@@ -90,128 +37,166 @@ async def test_full_config(hass):
-    config = {azure_event_hub.DOMAIN: {"filter": filter_config}}
-    config[azure_event_hub.DOMAIN].update(MIN_CONFIG)
-
+async def test_filter_only_config(hass):
+    """Test the popping of the filter and further import of the config."""
+    config = {
+        azure_event_hub.DOMAIN: {
+            "filter": {
+                "include_domains": ["light"],
+                "include_entity_globs": ["sensor.included_*"],
+                "include_entities": ["binary_sensor.included"],
+                "exclude_domains": ["light"],
+                "exclude_entity_globs": ["sensor.excluded_*"],
+                "exclude_entities": ["binary_sensor.excluded"],
+            },
+        }
+    }
     assert await async_setup_component(hass, azure_event_hub.DOMAIN, config)
-    await hass.async_block_till_done()
-    mock_call_later.assert_called_once()
-    return mock_call_later.call_args[0][2]
 
 
-async def _run_filter_tests(hass, tests, process_queue, mock_batch):
-    """Run a series of filter tests on azure event hub."""
+async def test_unload_entry(hass, entry):
+    """Test being able to unload an entry."""
+    assert await hass.config_entries.async_unload(entry.entry_id)
+    assert entry.state == ConfigEntryState.NOT_LOADED
+
+
+async def test_failed_test_connection(hass, mock_get_eventhub_properties):
+    """Test being able to unload an entry."""
+    entry = MockConfigEntry(
+        domain=azure_event_hub.DOMAIN,
+        data=SAS_CONFIG_FULL,
+        title="test-instance",
+        options=BASIC_OPTIONS,
+    )
+    entry.add_to_hass(hass)
+    mock_get_eventhub_properties.side_effect = EventHubError("Test")
+    try:
+        await hass.config_entries.async_setup(entry.entry_id)
+    except azure_event_hub.ConfigEntryNotReady:
+        pass
+    assert entry.state == ConfigEntryState.SETUP_RETRY
+
+
+async def test_stop(hass, hub):
+    """Test stopping the hub, which empties the queue."""
+    assert await hub.async_stop()
+    assert hub.queue.empty()
+
+
+async def test_send_batch_error(hass, hub, mock_send_batch):
+    """Test a error in send_batch."""
+    mock_send_batch.reset_mock()
+    mock_send_batch.side_effect = EventHubError("Test")
+    await hub.async_send(None)
+    mock_send_batch.assert_called_once()
+
+
+async def test_late_event(hass, hub, mock_create_batch):
+    """Test the check on late events."""
+    with patch(
+        f"{AZURE_EVENT_HUB_PATH}.time.monotonic",
+        return_value=monotonic() + timedelta(hours=1).seconds,
+    ):
+        await hub.async_send(None)
+        mock_create_batch.add.assert_not_called()
+
+
+async def test_full_batch(hass, hub, mock_create_batch):
+    """Test the full batch behaviour.
+
+    Can't use async_send, because that causes a loop with this side_effect.
+    """
+    mock_create_batch.add.side_effect = ValueError
+    assert hub.queue.qsize() == 1
+    async with hub._client.client as client:  # pylint: disable=protected-access
+        await hub.fill_batch(client)
```

Comment:
To test the hub we want to fire events on the Home Assistant event bus and then check that the mocked library is called correctly.

Reply:
We should only patch the library. To have different sequential side effects we can set a list of side effects as side effect on the mock.

Reply:
Nice, I didn't know that was possible!